pkg/internalapis/cnsoperator/triggercsifullsync/v1alpha1/register.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// SchemeGroupVersion define schema Group and version
+var SchemeGroupVersion = schema.GroupVersion{
+	Group:   "cns.vmware.com",
+	Version: "v1alpha1",
+}
+
+var (
+	schemeBuilder      runtime.SchemeBuilder
+	localSchemeBuilder = &schemeBuilder
+	// AddToScheme helps add all the stored functions to the scheme
+	AddToScheme = localSchemeBuilder.AddToScheme
+)
+
+func init() {
+	// We only register manually written functions here. The registration of the
+	// generated functions takes place in the generated files. The separation
+	// makes the code compile even when the generated files are missing.
+	localSchemeBuilder.Register(addKnownTypes)
+}
+
+// Resource takes an unqualified resource and returns a Group qualified GroupResource
+func Resource(resource string) schema.GroupResource {
+	return SchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+// Adds the list of known types to the given scheme.
+func addKnownTypes(scheme *runtime.Scheme) error {
```

Comment:
All of this is already being done here - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/internalapis/register.go#L77. 

I don't think you need any of this.

https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/kubernetes/kubernetes.go#L186

https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/internalapis/register.go#L77

are already doing this.

Reply:
I missed that file while looking. Let me remove the one I added and reuse that instead.

Reply:
@BaluDontu on a closer look, https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/kubernetes/kubernetes.go#L186 is only adding those Kinds to a local scheme which is used to create a k8s client. 

Registering the Kind with the controller manager's scheme is required for the bug to be fixed. 

`register.go` can indeed be removed. 

Please see updated logs in the description.

Reply:
Sure, makes sense.